### PR TITLE
fix(core): Register workflows as active only after all of the triggers and pollers setup successfully

### PR DESCRIPTION
**Pull request by netroy**
## Summary
`ActiveWorkflows` is a key component in n8n's execution engine, but so far has had 0% test coverage.
This PR gets the line/statement coverage for this file to close to 100%. 

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [x] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/core/src/__tests__/ActiveWorkflows.test.ts [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+					activeWorkflows.add(
+						workflowId,
+						workflow,
+						additionalData,
+						mode,
+						activation,
+						getTriggerFunctions,
+						getPollFunctions,
+					),
+				).rejects.toThrow(WorkflowActivationError);
```
Comment (tomi) [inline packages/core/src/__tests__/ActiveWorkflows.test.ts:121]: Should we check that `isActive` returns false?

File: packages/core/src/__tests__/ActiveWorkflows.test.ts [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
```
Comment (tomi) [inline packages/core/src/__tests__/ActiveWorkflows.test.ts:143]: Same here

File: packages/core/src/__tests__/ActiveWorkflows.test.ts [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+				getTriggerFunctions,
+				getPollFunctions,
+			);
+
+			await expect(activeWorkflows.remove(workflowId)).rejects.toThrow(
+				`Failed to deactivate trigger of workflow ID "${workflowId}": "Close function failed"`,
+			);
+
+			expect(triggerResponse.closeFunction).toHaveBeenCalled();
+			expect(errorReporter.error).not.toHaveBeenCalled();
```
Comment (tomi) [inline packages/core/src/__tests__/ActiveWorkflows.test.ts:305]: Should `isActive` be true or false now?

**Review by tomi — DISMISSED** [review]
Nice work 👏 Couple minor suggestions

File: packages/core/src/__tests__/ActiveWorkflows.test.ts (hunk 1 above)
Comment (netroy) [reply packages/core/src/__tests__/ActiveWorkflows.test.ts:121]: good catch. adding `expect(activeWorkflows.isActive(workflowId)).toBe(false)` here breaks the test. I think we might have caught a bug here,

File: packages/core/src/__tests__/ActiveWorkflows.test.ts (hunk 3 above)
Comment (netroy) [reply packages/core/src/__tests__/ActiveWorkflows.test.ts:305]: it's actually unclear. I've posted [here](https://n8nio.slack.com/archives/C069HS026UF/p1734530918502139) for clarification.  
Once we have consensus, we can create a new ticket and new PR to address this.

File: packages/core/src/__tests__/ActiveWorkflows.test.ts [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+						workflow,
+						additionalData,
+						mode,
+						activation,
+						getTriggerFunctions,
+						getPollFunctions,
+					),
+				).rejects.toThrow(WorkflowActivationError);
+				expect(activeWorkflows.isActive(workflowId)).toBe(false);
+			});
```
Comment (tomi) [inline packages/core/src/__tests__/ActiveWorkflows.test.ts:144]: We could have a test case for a WF that has both pollers and triggers

File: packages/core/src/ActiveWorkflows.ts [hunk 5]
```diff
@@ -104,6 +98,8 @@ export class ActiveWorkflows {
 			}
 		}
 
+		this.activeWorkflows[workflowId] = { triggerResponses };
```
Comment (tomi) [inline packages/core/src/ActiveWorkflows.ts:101]: Since this changes the behaviour and potentially fixes a bug in the activation, should we change the PR title to `fix(core)` so it's included in the changelog? Or we could also extract this to a separate PR

**Review by tomi — DISMISSED** [review]
Couple suggestions but otherwise LGTM 👍

File: packages/core/src/ActiveWorkflows.ts (hunk 5 above)
Comment (netroy) [reply packages/core/src/ActiveWorkflows.ts:101]: Updated the title. I don't think a separate PR is necessary here.

**Review by tomi — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.74.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.74.1)

**Comment by janober** [comment]
Got released with [`n8n@1.74.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.74.0)

**Comment by janober** [comment]
Got released with [`n8n@1.74.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.74.0)
